### DISCUSSION TITLE
Handle AADSTS50105 error with Azure AD

### DISCUSSION THREAD

**Discussion by Laurentvw** [nextauthjs/next-auth#5457] [Help]
### Question 💬

I am using the Azure AD provider to sign in a user. When the user does not have the required access, I end up on a Microsoft error page (https://login.microsoftonline.com/common/SAS/ProcessAuth) with the following error:

> AADSTS50105: Your administrator has configured the application XXX ('X') to block users unless they are specifically granted ('assigned') access to the application. The signed in user 'X' is blocked because they are not a direct member of a group with access, nor had access directly assigned by an administrator. Please contact your administrator to assign access to this application.

I would like to handle this error in the web app by having it redirect to a custom error page. I have tried a couple of things but I am not able to make it redirect back to the web app.

The callbacks `signIn ` and `redirect` do not seem to be triggered after the failed sign in attempt. I have tried a couple of things:

```
export const authOptions: NextAuthOptions = {
  providers: [
    AzureADProvider({...}),
  ],
  pages: {
    error: '/forbidden', // This doesn't work.
  },
  callbacks: {
    async signIn({ user, account, profile, email, credentials }) {
      const isAllowedToSignIn = false; // test
      if (isAllowedToSignIn) {
        return true
      } else {
        return '/forbidden'  // This doesn't work.
      }
    },
    async redirect({ url, baseUrl }) {
      console.log('redirect'); // Not triggered either
      return baseUrl
    },
  }
}
```
Anyone know how to make this work? Thank you.

### How to reproduce ☕️

See above.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by vishnu-valiveti** [comment]
Any workaround to add the custom message on this page?
